### ISSUE TITLE
AEM 6.3 compatibility/verification

### ISSUE THREAD

**Issue by jbornemann** [TWCable/grabbit#185]
@sagarsane

**Comment by sagarsane** [comment]
👍

**Comment by dscirto** [comment]
I'm having issues integrating Grabbit 7.1.2 on a new AEM 6.3. Long story short, any time I perform a GET on /grabbit path (eg. localhost:4502/grabbit/job), I always run into this:

![image](https://user-images.githubusercontent.com/22640713/27036874-4080209c-4f54-11e7-8cc2-361fc0a6431d.png)

I notice I also receive a 409 response on PUT requests. Anyone else running into this on 6.3? Or is there a configuration I'm missing? Did not have any issues with previous versions on 6.2, so I'm glomping on this open issue.

**Comment by jbornemann** [comment]
Interesting.

We might have to do some debugging in the GrabbitResourceProvider

**Comment by dscirto** [comment]
Thanks @jbornemann , let me know if there are any configurations I should look at that could be indirectly causing problems or if you need some logs down the road to help with debugging.

**Comment by sagarsane** [comment]
@dscirto just curious .. you don't have the Deserialization firewall enabled on your instance do you?

If you do, you may need to install the Config Package provided to exclude Grabbit from it.

More info can be found here : https://github.com/TWCable/grabbit/blob/master/docs/AEMSupport.adoc#what-you-need-to-run-grabbit (the 3rd bullet point)

**Comment by viveksachdeva** [comment]
@sagarsane, @jbornemann  : I started fresh, installed all the config packages.. Running into multiple issues..
- For new job request, config JSON that works on 6.1 not working on 6.3
Processing.....

~~~~~~~~~Failure~~~~~~~~~~~~~

Something went wrong when processing the given config.

cURL, the transfer library used by this shell received an error code of 22. See https://curl.haxx.se/libcurl/c/libcurl-errors.html for debugging.
If this doesn't help, please check the client log for more details.

- For monitor job as well, its not going to the GrabbitJobServlet..

I saw difference in behavior of Resource provider between 6.3 and 6.1. Taking example of monitor job, on:
- 6.1, getResource is called 3 times - /grabbit/job/dsdads, /grabbit/job and /grabbit.. Then it goes to the GrabbitJobServlet.
- 6.3, getResource is called 3 times - /grabbit/job/dsdads, /grabbit/job and /grabbit.. This goes to RootResourceServlet..

**Comment by viveksachdeva** [comment]
One more observation:
- /grabbit/job/adsa.json resolves correctly to the resource but not to the servlet in 6.3
<img width="1111" alt="screen shot 2017-06-13 at 4 04 41 pm" src="https://user-images.githubusercontent.com/1879953/27072146-0cc16780-5052-11e7-82ea-e5f8635f2864.png">
<img width="872" alt="screen shot 2017-06-13 at 4 04 36 pm" src="https://user-images.githubusercontent.com/1879953/27072147-0cc2a758-5052-11e7-8f0e-94a0433f7463.png">

**Comment by dscirto** [comment]
@sagarsane Looks like OOTB AEM 6.3 has Deserialization Firewall configured so I'll take your suggestion

**Comment by sagarsane** [comment]
@jbornemann @dscirto @viveksachdeva I think its happening because of this #188

**Comment by dscirto** [comment]
Will try to get to some testing today and will look at #188

**Comment by dscirto** [comment]
I can verify seeing the same thing as @viveksachdeva 

Noticed in relation to #188 that checking "Bypass the whitelist" config option does not resolve issue. Note that I have not run across the exception in #188 however, with-or-without the configuration option

**Comment by sagarsane** [comment]
@dscirto just to clarify -- I saw the exception mentioned in #188 when the bundle is being installed and activated.

**Comment by viveksachdeva** [comment]
@sagarsane : Checked by whitelisting config mentioned in #188.. The error goes away with that.. Whitelisting it could be a stopgap fix, so atleast grabbit works on 6.3..

**Comment by dscirto** [comment]
@viveksachdeva does that mean you have been able to trigger a sync with the whitelisting?

**Comment by viveksachdeva** [comment]
@dscirto : yeah.. I can sync content by after making that config change

**Comment by dscirto** [comment]
I was able to verify that I can sync from one local AEM 6.3 to another local AEM 6.3 with the firewall deserialization package and login admin whitelist. I am also able to view jobs as expected at "/grabbit/job/all.json" as expected.

Thanks for the debugging!

**Comment by sagarsane** [comment]
Thanks for confirming @dscirto and for raising this issue!

**Comment by remingtonreed** [comment]
I'm still having the issues described above with Grabbit 7.1.2 on AEM 6.3. I have installed the Deserialization firewall package as well as checked whitelist.bypass on org.apache.sling.jcr.base.internal.LoginAdminWhitelist.

Admittedly, this is my first time using Grabbit, so it could be something else I'm doing. Any suggestions?

**Comment by viveksachdeva** [comment]
@remingtonreed : Could you please share the exact issue that you are having..? Logs would be helpful..

**Comment by remingtonreed** [comment]
I'm pretty well getting the same results from above. Here is the message that the bash script returns.

```
~~~~~~~~~Failure~~~~~~~~~~~~~

Something went wrong when processing the given config.

cURL, the transfer library used by this shell received an error code of 22. See https://curl.haxx.se/libcurl/c/libcurl-errors.html for debugging.
If this doesn't help, please check the client log for more details.
```

I've checked the grabbit-receive.log, grabbit-send.log, and error.log of both servers and can't see any meaningful error, so I'm not sure what else to check there.

Here are the OSGi configs after modifying them (this is what both the receiving and sending end look like)

![admin-whitelist](https://user-images.githubusercontent.com/403897/29420206-8e70607e-832e-11e7-9009-ad0e530a3098.png)

![deserialization-firewall](https://user-images.githubusercontent.com/403897/29420210-934b2318-832e-11e7-8a50-9299cee9b617.png)

Unfortunately pretty well all of the useful pieces of the config seem like sensitive information, however, I just copied the sample one and modified each field. Thanks!

**Comment by viveksachdeva** [comment]
@remingtonreed : Did you install Sun-Misc-Fragment bundle as well as mentioned in README..
https://bintray.com/artifact/download/twcable/aem/dependencies/Sun-Misc-Fragment-Bundle-1.0.0.zip

**Comment by remingtonreed** [comment]
@viveksachdeva: Yes, here are the packages I installed on both sending and receiving instances.

![image](https://user-images.githubusercontent.com/403897/29444597-2b04629a-839e-11e7-8a6e-c906c06c2023.png)

**Comment by bobbymavrov** [comment]
Hey @remingtonreed, I'm facing the exact same issues right now. Were you able to get any progress? 

I have the three packages installed and the whitelist.bypass checked but I see the cURL error code 22. 

Thanks a lot!

**Comment by henryleunginm** [comment]
Hi,

I'm having the same issue in AEM 6.3 as described by @remingtonreed and @bobbymavrov. Have you guys managed to solve the issue?

Thanks!

**Comment by harish6959** [comment]
Hi,

I'm having the same issue in AEM 6.3 as described by @remingtonreed.

Thanks in advance for any suggestion and fix.

**Comment by a41pavan** [comment]
@remingtonreed What do you get when you do this?

/system/console/healthcheck?tags=deserialization

**Comment by Lakshmivl** [comment]
i have the same issue. Installed all the 3 packages as mentioned above and bypassed whilte list.
recieving cURL error code 22. need help 

output of /system/console/healthcheck?tags=deserialization

![image](https://user-images.githubusercontent.com/39534592/40384492-b727c0b0-5dd1-11e8-88e5-f538a574a470.png)

**Comment by masroormohammed** [comment]
Had to make an additional config update to make sure grabbit worked(to overcome the not whitelisted issue)
<img width="1303" alt="screen shot 2018-05-30 at 3 55 38 pm" src="https://user-images.githubusercontent.com/10712665/40744382-55d96b4a-6422-11e8-852a-85ba87992018.png">
